### PR TITLE
Clean up import rendering in codegen.

### PR DESCRIPTION
**Pull request by lthibault**
This is a small refactor that aims for an incremental improvement to `fileparts.go`'s readability.

I had to stare at the `imports` struct a wee bit longer than I would have liked, owing to the indirection in its initialization.  This PR removes the `imports.init` method and replaces it with a simple `newImports` constructor.  Runtime behavior is unaffected.

⚠️ N.B.:  this is branched off of #285, so merge that one first.

### CODE REVIEW

**Comment by zenhack** [comment]
Hm, this still has the commits from #285 on it, even though that's merged... makes the diff hard to follow...

**Comment by lthibault** [comment]
Rats.  I'll rebase and push.

**Comment by lthibault** [comment]
Fixed.

**Comment by lthibault** [comment]
Bump.  Is this okay to merge?

File: capnpc-go/fileparts.go [hunk 1]
```diff
@@ -140,6 +149,10 @@ func (i *imports) byName(name string) (spec importSpec, ok bool) {
 }
 
 func (i *imports) add(spec importSpec) (name string) {
+	if i.used == nil {
+		i.used = make(map[string]bool)
```
Comment (zenhack) [inline capnpc-go/fileparts.go:153]: Why do this here, as opposed to just putting it at the top of newImports?

Comment (lthibault) [reply capnpc-go/fileparts.go:153]: No particular reason.  I think this is an artifact from the refactoring where there was some lazy initialization going on.

I'll make the change.

Comment (lthibault) [reply capnpc-go/fileparts.go]: Fixed.

**Review by zenhack — APPROVED** [review]
